### ISSUE TITLE
Completely unrelated emails being threaded together

### ISSUE THREAD

**Issue by ceribik** [protz/GMail-Conversation-View#904]
Hopefully this will make sense.

I received email A from e@e.com a few weeks ago which was sent my email c@c.com.

I just received another email B from YouTube (completely unrelated to e@e.com) which was sent to foo@pages.plusgoogle.com which arrived in the same inbox as c@c.com. This email does not have a header for "In-Reply-To:" or "References:"

Bug: both messages (email A and email B) were threaded together, with email B being a child of email A.

Note that I'm using IMAP for c@c.com

Any ideas on what is causing this?

**Comment by bastienho** [comment]
I have the same issue. I've checked both "in-reply-to" and "references" and there is nno relation.

it seems that there is bit between TB and TBC as explained (here)[https://github.com/protz/GMail-Conversation-View/blob/a2a06d1e2af43fabcbb48a41952312ef7ce3b8d5/modules/conversation.js#L357]:
```javascript
 // Ok, interesting bit. Thunderbird has that non-strict threading thing, i.e.
// it will thread messages together if they have a "Green Llama in your car"
// "Re: Green Llama in your car" subject pattern, and EVEN THOUGH they do not
// have the correct References: header set.
// Until 2.0alpha2, what we would do is:
// - fetch the Gloda message collection,
// - pick the first Gloda message, get the message collection for its
// underlying conversation,
// - merge the results for the conversations with the initially selected set,
// - re-stream all other messages except for the first one, because we only
// have their nsIMsgDbHdr.
// That's sub-optimal, because we actually have the other message's Gloda
// representations at hand, it's just that because the headers do not set the
// threading, gloda hasn't attached them to the first message.
// The solution is to merge the initial set of messages, the gloda messages
// corresponding to the intermediate query, and the initially selected
// messages...
```

**Comment by pianoslum** [comment]
There's even a FAQ entry: https://github.com/protz/GMail-Conversation-View/wiki/What-is-threading

**Comment by protz** [comment]
The comment is indeed accurate, but what you're experiencing is most likely the bug pointed to by the FAQ entry.

Thanks,

Jonathan

**Comment by AndreC10002** [comment]
It is sad, but I have this same problem 9 years after this thread was opened. I'm running 115.10.1 on Win 10. All was good, until I undeleted a message. After that, threads are all messed up. No, it wasn't somebody replying with a changed subject. It was nobody's fault, but TB somehow deciding to put together absolutely unrelated messages after an undelete.

I'm a TB user for as long as I can remember. Many years really. I used to love it. But I have to be honest, in the last 1-2 years TB isn't getting better. Quite the opposite, it is declining. Bugs are returning, people are complaining about the UI, it is slower. In the morning it takes 10-15 minutes to get my new emails from 4 mailboxes. It wasn't like that versions ago.
I'm not sure what is going on with TB. I wish I could roll back to a version from 2 years ago and stick with it.  I'm about to look for alternatives, unfortunately.

I don't have much faith somebody will review this ticket and fix the bug, but here it is for the record.

**Comment by AndreC10002** [comment]
It is sad, but I have this same problem 9 years after this thread was opened. I'm running 115.10.1 on Win 10. All was good, until I undeleted a message. After that, threads are all messed up. No, it wasn't somebody replying with a changed subject. It was nobody's fault, but TB somehow deciding to put together absolutely unrelated messages after an undelete.

I'm a TB user for as long as I can remember. Many years really. I used to love it. But I have to be honest, in the last 1-2 years TB isn't getting better. Quite the opposite, it is declining. Bugs are returning, people are complaining about the UI, it is slower. In the morning it takes 10-15 minutes to get my new emails from 4 mailboxes. It wasn't like that versions ago.
I'm not sure what is going on with TB. I wish I could roll back to a version from 2 years ago and stick with it.  I'm about to look for alternatives, unfortunately.

I don't have much faith somebody will review this ticket and fix the bug, but here it is for the record. BTW, 115.12.1 doesn't fix the issue.
